src/_pytest/assertion/util.py

```diff
@@ -319,6 +329,41 @@ def _compare_eq_dict(left, right, verbose=False):
     return explanation
 
 
+def _compare_eq_class(left, right, verbose, type=None):
+    if type == "data":
```

Comment:
lets please change the magical strings to named functions that do the extraction and always require passing them